File: functions/libsodium/variants/go/4_fBlaMka_8.go
package main

func fBlaMka(x, y uint64) uint64 {
	const mask uint64 = 0xFFFFFFFF
	xy := (x & mask) * (y & mask)
	return x + y + (xy * 2)
}


func main(){
    fBlaMka(0,0)
}

File: functions/libsodium/variants/c/4_fBlaMka_8.c
#include <stdint.h>

static inline uint64_t fBlaMka(uint64_t x, uint64_t y)
{
    uint32_t low_x = (uint32_t)(x & UINT64_C(0xFFFFFFFF));
    uint32_t low_y = (uint32_t)(y & UINT64_C(0xFFFFFFFF));
    uint64_t product = (uint64_t)low_x * low_y;
    return x + y + (product << 1);
}


int main(){
    fBlaMka(0,0);
}
